src/main/groovy/org/shipkit/internal/notes/contributors/AllContributorsFetcher.java

```diff
@@ -40,16 +43,36 @@ ProjectContributorsSet fetchAllContributorsForProject(String repository, String
         return result;
     }
 
-    private Set<ProjectContributor> extractContributors(List<JsonObject> page, String readOnlyAuthToken) throws IOException, DeserializationException {
+    private Set<ProjectContributor> extractContributors(List<JsonObject> page, final String readOnlyAuthToken) throws IOException, DeserializationException {
         Set<ProjectContributor> result = new HashSet<ProjectContributor>();
-        for (JsonObject contributor : page) {
-            //Since returned contributor does not have 'name' element, we need to fetch the user data to get his name
-            //TODO add static caching of this. Names don't change that often, let's just cache this forever in build cache.
-            String url = (String) contributor.get("url");
-            GitHubObjectFetcher userFetcher = new GitHubObjectFetcher(url, readOnlyAuthToken);
-            JsonObject user = userFetcher.getPage();
-            result.add(GitHubAllContributorsJson.toContributor(contributor, user));
+        //Since returned contributor does not have 'name' element, we need to fetch the user data to get his name
+        //TODO add static caching of this. Names don't change that often, let's just cache this forever in build cache.
+        ExecutorService executor = Executors.newFixedThreadPool(4);
+        int maxSizePerCallable = 25;
+
+        List<Future<Set<ProjectContributor>>> futures = new ArrayList<Future<Set<ProjectContributor>>>();
+        if (page.size() > 0) {
+            for (int i = 0; i < Math.max(page.size() / maxSizePerCallable, 1); i++) {
+                List<JsonObject> subList = page.subList(i * maxSizePerCallable, Math.min((i + 1) * maxSizePerCallable, page.size()));
+
+                GitHubObjectFetcher objectFetcher = new GitHubObjectFetcher(readOnlyAuthToken);
+                Function<JsonObject, ProjectContributor> projectContributorFetcherFunction = new ProjectContributorFetcherFunction(objectFetcher);
+                Callable<Set<ProjectContributor>> callable = new FetcherCallable<JsonObject, ProjectContributor>(subList, projectContributorFetcherFunction);
+
+                futures.add(executor.submit(callable));
+            }
         }
+
+        for (Future<Set<ProjectContributor>> future: futures) {
+            try {
+                result.addAll(future.get());
+            } catch (InterruptedException e) {
```

Comment:
For simplicity, I suggest single catch(Exception) followed by throwing a RuntimeException with a decent message